src/main/java/com/thed/utils/XMLLink.java

```diff
@@ -59,6 +59,9 @@ public void attachLink(String[] linkArray, Type type) {
                 }
                 xmlLink.attachLink(Arrays.copyOfRange(linkArray, 1, linkArray.length), type);
             } else {
+                if(linkArray.length == 2) {
```

Comment:
what does linkArray contains?

Reply:
For a xml variable $testsuite.testcase.failure, link array will be ["testsuite", "testcase", "failure"]. These are links that we will traverse down while parsing.